### PR TITLE
chore: add error throwing utility

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

This PR introduces a new error utility that is intended to solve a problem present in varying places around the codebase - namely, passing `mate::Arguments*` for no other purpose than to throw errors to JS. This aims to solve that problem with a new `util::ErrorThrower` class, whose sole purpose is to throw one of a set of descriptive errors, and to do that **only**. I included a proof-of-concept conversion to this new pattern in `SystemPreferences::GetSystemColor`, which is tested to work on macOS. 

It increases ease of readability for parameters, since it's often unclear _why_ we're passing `mate::Arguments*` to a given function since `mate::Arguments*` is basically analogous to 'anything'. In a follow-up PR, I plan to introduce the ability to return a thrown error so that we no longer need to do:

```cpp
thrower.ThrowError("oh no you did a bad thing");
return std::string();
```

and can instead cleanly reduce LoC with:

```cpp
return thrower.ThrowError("oh no you did a bad thing");
```

independent of the caller fn's return type.

cc @ckerr @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: shell/common/error_util.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// A wrapper around V8ThrowException.
+
+class Error {
+ public:
+  Error(v8::Isolate* isolate);
+  Error();
+
+  ~Error();
+
+  v8::Isolate* isolate() const { return isolate_; }
```
Comment (MarshallOfSound) [inline shell/common/error_util.h]: Does this have to be public?

File: shell/common/error_util.cc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#include "native_mate/converter.h"
+#include "shell/common/error_util.h"
+
+namespace electron {
+
+namespace util {
+
+Error::Error(v8::Isolate* isolate) : isolate_(isolate) {}
+
+// This constructor should be rarely if ever used, since
```
[the remark this answers is not in the document]
Comment (codebytere) [reply shell/common/error_util.cc]: ah fair, will do!

File: shell/common/error_util.cc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#include <string>
+
+#include "native_mate/converter.h"
+#include "shell/common/error_util.h"
+
+namespace electron {
+
+namespace util {
+
+Error::Error(v8::Isolate* isolate) : isolate_(isolate) {}
```
[the remark this answers is not in the document]
Comment (codebytere) [reply shell/common/error_util.cc]: ErrorThrower seems apt for now although that may change on second iteration 🤔

File: shell/common/error_util.h (hunk 1 above)
Comment (codebytere) [reply shell/common/error_util.h]: no, will move!

**Pull request merged**